test/unit/api/api-test.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const _ = require('lodash');
+const sinon = require('sinon');
+const proxyquire = require('proxyquire');
+
+const ctx = {http: {}, client: {}};
+const ctxStub = sinon.stub().returns(ctx);
+const methodSpies = {};
+const allSpy = WidgetStub.all = sinon.spy();
+const constructorSpy = sinon.spy();
+
+function WidgetStub (opts, ctx) {
+  constructorSpy(opts, ctx);
+  _.extend(this, methodSpies);
+}
+
+const api = proxyquire('../../..', {
+  './widget': WidgetStub,
+  './context': ctxStub
+});
+
+describe('API', function () {
+  describe('initialization', function () {
+    it('passes token and host to context creation fn', function () {
+      api.createClient({
+        accessToken: 'token-lol',
+        host: 'http://api.test.com'
+      });
+
+      sinon.assert.calledOnce(ctxStub.withArgs({
+        token: 'token-lol',
+        host: 'http://api.test.com'
+      }));
+    });
+  });
+
+  describe('instance methods', function () {
+    beforeEach(function () {
+      methodSpies.read = sinon.spy();
+      methodSpies.save = sinon.spy();
+      methodSpies.delete = sinon.spy();
+      this.client = api.createClient({spaceId: 'spaceid'});
+    });
+
+    it('#getAll()', function () {
+      this.client.getAll();
+      sinon.assert.calledOnce(allSpy.withArgs({spaceId: 'spaceid'}));
```

Comment:
The meaning `allSpy` is not very obvious. Why not use `WidgetStub.all` directly?